docs/design/executors/01-executor-service-stats.md

```diff
@@ -0,0 +1,100 @@
+# Scheduled and Durable Executor Service Statistics
+
+|ℹ️ Since: 4.1|
+|-------------|
+
+## Background
+
+### Description
+
+Hazelcast has 3 different executor service implementations exposed with 
+public api. These three different types of executor services are called
+as executor service, scheduled executor service and durable executor 
+service. Only executor type which has statistics available 
+among these is executor service. Statistics collection is not done for 
+scheduled and durable ones.
+
+
+## Design
+
+In this design, we make same statistics with executor service available 
+for the other executor types. These statistics are only available for
+monitoring from management center. The statistics we have currently is 
+represented with class `LocalExecutorStats`.
+
+```
+public interface LocalExecutorStats extends LocalInstanceStats {
+
+    /**
+     * Returns the number of pending operations on the executor service.
+     *
+     * @return the number of pending operations on the executor service
+     */
+    long getPendingTaskCount();
+
+    /**
+     * Returns the number of started operations on the executor service.
+     *
+     * @return the number of started operations on the executor service
+     */
+    long getStartedTaskCount();
+
+    /**
+     * Returns the number of completed operations on the executor service.
+     *
+     * @return the number of completed operations on the executor service
+     */
+    long getCompletedTaskCount();
+
+    /**
+     * Returns the number of cancelled operations on the executor service.
+     *
+     * @return the number of cancelled operations on the executor service
+     */
+    long getCancelledTaskCount();
```

Comment:
By the way, if I am not missing something, the canceled task count is only incremented for `ExecutorService`. For these two, does this metric mean anything?

Reply:
good catch, i have added it to scheduled one, not sure about durable one leaving it as TODO for now.